lib/assessment.js

```diff
@@ -24,7 +30,11 @@ var sum = function sum() {
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
-
+  result = true;
+  array.forEach(function(a){
+    result = result && callback(a);
+  });
+  return result;
```

Comment:
This is an `every` function, not an `any` function. Here is an explanation of `any` versus `every`, and the steps needed to convert `every` to `any`:

The question asks for a function that returns `true` if the callback returns `true` for **any** element in the array. This suggests that we should be using a logical OR (`||`), not a logical AND (`&&`). A logical AND would be suitable for determining whether a callback returns `true` for **all** elements of the array. 

To know what our initial value of `result` should be, we consult the [truth table for the OR operation](https://en.wikipedia.org/wiki/Logical_disjunction#Truth_table). As we see on row 2 of the truth table, `true || false` evaluates to `true`. Therefore, if we initialize `result` to `true`, `any` will return `true` regardless of what the callback returns for each element of the list -- this is not what we want. There's only one more value to consider: `false`.

We see on lines 4 and 3 of the truth table that `false || false` yields `false`, and `false || true` yields `true`. So, if `result` starts out as `false`, it will remain `false` until it is ORed with `true`. This sounds like what we want. Also, once `result` stores `true`, it will remain `true` no matter what we OR it with. This also sounds like what we want here.

So line 42 should read: `var result = false;` and line 45 should read: `result = result || callback(...);`

3/4 credit.